cmd/webhook.go

```diff
@@ -171,6 +171,7 @@ func init() {
 	rootCmd.AddCommand(webhookCmd)
 	webhookCmd.Flags().StringVar(&metricsAddr, "metrics-addr", ":8080", "The address the metric endpoint binds to.")
 	webhookCmd.Flags().StringVar(&healthzAddr, "healthz-addr", ":8081", "The address the health endpoint binds to.")
+	webhookCmd.Flags().IntVar(&port, "port", 443, "The address the health endpoint binds to.")
```

Comment:
Actually, i would like to avoid a breaking change. As users need to allow the control-plane to access the pods through a security group #956.
Do you know if port 443/ingress is always open on EKS/GKE nodes? If so i'm fine with using `443`.

Reply:
FYI: I noticed in #981 that EKS + GKE `managed` e2e tests fail due to the webhook issue

Reply:
GKE: Yes - see here https://cloud.google.com/kubernetes-engine/docs/how-to/private-clusters#add_firewall_rules

EKS: they do reference minimal security groups with 443 port: https://docs.aws.amazon.com/eks/latest/userguide/sec-group-reqs.html#control-plane-worker-node-sgs I think this is the default for the terraform modules, and I think users might need to have this configured in order to have a functional cluster as per this link.l

Reply:
Although for EKS it is not clear to me if this is the node port or the pod port.

Reply:
Confirmed for eks+terraform 443 is enabled by default: https://github.com/terraform-aws-modules/terraform-aws-eks/blob/3c6686379de9489d4c044cd13a63e2329e2ae4e1/node_groups.tf#L78-L85

The only issue I see is that in some deployments the pod might not be authorized to run on port 443 due to it being a privileged port or to be already in use. I don't think we can cover both restrictions without a customization by the user.

IMO, I'm fine to let the port to 9443, as users could then either configure firewall rules or change the port to 443, according to their convenience.

Reply:
> FYI: I noticed in #981 that EKS + GKE `managed` e2e tests fail due to the webhook issue

This makes pointing to 443 kinda useless. It basically means that extra privilege would be needed for the webhook to start on port `443`.. the default should indeed be `9443`. One thing that I did for our e2e tests was this, though: https://github.com/external-secrets/external-secrets/pull/965/files#diff-1884f2cd2ea2154f37484f284e4a24197737dbd4c8d08d0ff1c346496c86237cR38-R41.

I'm putting the defaults to 9443

Reply:
> The only issue I see is that in some deployments the pod might not be authorized to run on port 443

The container would require `NET_BIND_SERVICE` capability.

Just for reference how cert-manager does it: they let the webhook pod listen run under `10250`. We should do that, too i think.

 https://github.com/cert-manager/cert-manager/blob/master/deploy/charts/cert-manager/values.yaml#L225-L232